aeppl/utils.py

```diff
@@ -32,6 +32,13 @@
 ]
 
 
+def logsumexp(x, axis=None, keepdims=True):
```

Comment:
This graph will be "double" optimized, unless you place the switch inside the sum. You can also leave the max trick out since it's a default aesara rewrite now.

Reply:
If it's rewriting that log-sum, then we probably need to make sure that doesn't cause problems&mdash;and do something about it, if it does.

Reply:
It doesn't cause problems, it just adds an unecessary extra switch because the returned graph still contains a `log(sum(exp(x)))` structure even if `x` is now `x - switch(isinf(max(x), 0, max(x))`

Reply:
You can either let aesara do the rewrite or explicitly return the stable form `max(x) + log(sum(switch(isinf(max(x)), np.exp(x), np.exp(x-max(x))`

Reply:
Maybe we should add a helper `logsumexp` to aesara? To avoid reinventing it everywhere

Reply:
Definitely